parsec/api/protocol/base.py

```diff
@@ -32,11 +30,11 @@ class MessageSerializationError(SerdePackingError, ProtocolError):
     pass
 
 
-def packb(data: Dict[str, object]) -> bytes:
+def packb(data: Mapping) -> bytes:  # type: ignore[type-arg]
     return _packb(data, MessageSerializationError)
 
 
-def unpackb(data: bytes) -> Dict[str, object]:
+def unpackb(data: bytes) -> dict:  # type: ignore[type-arg]
```

Comment:
Why are those changes necessary?

Reply:
Because `unpackb` can return dict with non-str as keys, so it should be `Dict[object, object]`, however it seems `str` is not considered `object` by mypy when used as argument :/

```
parsec/backend/app.py:395: error: Argument 1 to "packb" has incompatible type "Dict[str, str]"; expected "Mapping[object, object]"  [arg-type]
```

So we endup with a `def packb(data: Mapping) -> bytes:` signature style, which is still not accepted by mypy
```
parsec/api/protocol/base.py:33: error: Missing type parameters for generic type "Mapping"  [type-arg]
```

Last step is to suppress this error with the `# type: ignore[type-arg]`

The funny thing is `packb`/`unpackb` from `api.protocol.base` module only call `packb`/`unpackb` from `serde.packing` module which have unparametrized `Mapping` and `dict` as param/return values (but mypy is not as strict in this module)

So in the end I got bored of pretending typing has any value in those functions (given we manipulate arbitrary dicts and pass them to a function that specify it wants arbitrary dicts ^^), so I just `type: ignore` the hell out of this :smile: